docs/component-specs.md

```diff
@@ -0,0 +1,29 @@
+# Spec files for agent-managed components
```

Comment:
Really nice to see some developer-facing documentation here.  Thanks for adding it!

Reply:
I'm hoping to fill this file out a lot more as one of my ON week tasks 😄